libs/utils/include/utils/RefCountedInternPool.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (C) 2025 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef TNT_UTILS_REFCOUNTEDINTERNPOOL_H
+#define TNT_UTILS_REFCOUNTEDINTERNPOOL_H
+
+#include <utils/Slice.h>
+#include <utils/FixedCapacityVector.h>
+#include <utils/Hash.h>
+#include <utils/Panic.h>
+#include <utils/debug.h>
+
+#include <tsl/robin_map.h>
+
+namespace utils {
+
+/** A reference-counted intern pool of slices of T. */
+template<typename T, typename Hash = std::hash<T>>
+class RefCountedInternPool {
```

Comment:
There is a lot of code in this class and everything is inlined. I would like to see if there is a way to use the inlines for type safety but have the "heavy" methods be in a cpp, acquire() and release() are going to generate tons of code.

Also unless we have multiple uses of this, maybe we keep it inside filament instead of libutils initially? 

I can't immediately think of a better name (naming is hard!) but I don't like "RefCounted" because it's an abbreviation.  Do we need to qualify it with "RefCounted" would "InternPool" be enough?